ponycheck/generator.pony

```diff
@@ -140,12 +140,92 @@ primitive Generators
         fun generate(rnd: Randomness): Set[T]^ =>
           let size = rnd.usize(0, max)
           let set = Set[T](size)
-          for i in Range(0, size) do
-            set.set(gen.generate(rnd))
-          end
+          set.union(
+            Iter[T^](gen.iter(rnd))
+              .take(size))
+          consume set
```

Comment:
This could be replaced with:
```pony
set .> union(
  Iter[T^](gen.iter(rnd))
    .take(size))
```
This is works for two reasons. First, the `.>` operator will send the receiver (`set`) forward so that the final value is type Set[T]. Second, any type with capability ref aliases as itself, so `A ref == A ref^`. Also as a side-note, the last value in a function body is automatically consumed without need for the `consume` keyword. I realize that these are a lot of small and usually inconsequential details of the Pony type system.